src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -0,0 +1,389 @@
+
+                        using (HttpResponseMessage response = await AzureHelper.RequestWithRetry(Log, client, createRequest))
+                        {
+                            if (response.IsSuccessStatusCode)
+                            {
+                                // Blobs can be files but have the name of a directory.  We'll skip those and log something weird happened.
+                                if (!string.IsNullOrEmpty(Path.GetFileName(filename)))
+                                {
+                                    Stream responseStream = await response.Content.ReadAsStreamAsync();
+
+                                    using (FileStream sourceStream = File.Open(filename, FileMode.Create))
+                                    {
+                                        responseStream.CopyTo(sourceStream);
+                                    }
+                                }
+                                else
+                                {
+                                    Log.LogWarning($"Unable to download blob '{blob}' as it has a directory-like name.  This may cause problems if it was needed.");
+                                }
+                            }
+                            else
+                            {
+                                Log.LogError("Failed to retrieve blob {0}, the status code was {1}", blob, response.StatusCode);
+                            }
+                        }
+
+                        string generatedIndex, downloadedIndex;
+                        //read current index.json
+                        using (StreamReader sr = new StreamReader(File.OpenRead(Path.Combine(IndexDirectory, relativeBlobPath, "index.json"))))
+                        {
+                            generatedIndex = sr.ReadToEnd();
+                        }
+
+                        //read downloaded index.json
+                        using (StreamReader sr = new StreamReader(File.OpenRead(Path.Combine(DownloadDirectory, relativeBlobPath, "index.json"))))
+                        {
+                            downloadedIndex = sr.ReadToEnd();
+                        }
+                        //deserialize
+                        List<NuGetVersion> generated = JsonConvert.DeserializeObject<IndexJson>(generatedIndex).ConvertToVersion;
+                        List<NuGetVersion> downloaded = JsonConvert.DeserializeObject<IndexJson>(downloadedIndex).ConvertToVersion;
+
+                        //merge
+                        foreach(var version in generated)
+                        {
+                            if (!downloaded.Contains(version))
+                            {
+                                downloaded.Add(version);
+                            }
+                        }
+                        downloaded.Sort();
+
+                        JsonSerializer jsonSerializer = new JsonSerializer();
```

Comment:
Why are we doing all this work in the scope of the HttpClient using block?

Reply:
removed.